crates/polars-utils/src/plpath.rs

```diff
@@ -96,50 +95,94 @@ impl<'a> fmt::Display for AddressDisplay<'a> {
     }
 }
 
-macro_rules! impl_scheme {
-    ($($t:ident = $n:literal,)+) => {
-        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
-        #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-        #[cfg_attr(feature = "dsl-schema", derive(schemars::JsonSchema))]
-        pub enum CloudScheme {
-            $($t,)+
-        }
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "dsl-schema", derive(schemars::JsonSchema))]
+pub enum CloudScheme {
+    Abfs,
+    Abfss,
+    Adl,
+    Az,
+    Azure,
+    File { has_hostname: bool },
```

Comment:
I think the smaller refactor here would be to add a `FileStripped` / `FileSpark` (or named otherwise) variant.